### ISSUE TITLE
Unclear from docs on namespace usage

### ISSUE THREAD

**Issue by bsambrone** [weaveworks/flux#814]
After speaking with Matthias, it appears that specifying the namespace in your yaml metadata is required. I didn't see anything in the docs about that. 

The situation is that I have a deployment in a namespace called dev (we don't use the default namespace). We have Weave cloud looking at GCR, and when I tried to release it told me that the service I wanted to update could not be found in the repository. It DID find the deployment of the same name that I had accidentally manually applied to the default namespace previously, but it was (correctly) ignored as the namespace was different.

This issue is for updating the docs to go into more detail about namespaces and how flux works with them. It also looks like filename and folder conventions are not needed.

**Comment by squaremo** [comment]
> This issue is for updating the docs to go into more detail about namespaces and how flux works with them. It also looks like filename and folder conventions are not needed.

Good call, thank you for pointing this out. We don't really talk about the requirements Flux has for the manifests, and it deserves a decent treatment. I shall sharpen my quill!
